app/src/main/java/com/teamwss/websoso/ui/onBoarding/OnBoardingViewModel.kt

```diff
@@ -0,0 +1,102 @@
+package com.teamwss.websoso.ui.onBoarding
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewmodel.initializer
+import androidx.lifecycle.viewmodel.viewModelFactory
+import com.teamwss.websoso.WebsosoApp
+import com.teamwss.websoso.domain.usecase.ValidateNicknameUseCase
+import com.teamwss.websoso.ui.onBoarding.first.model.NicknameInputType
+import com.teamwss.websoso.ui.onBoarding.first.model.OnBoardingFirstUiState
+import com.teamwss.websoso.ui.onBoarding.model.OnBoardingPage
+
+class OnBoardingViewModel(
+    private val validateNicknameUseCase: ValidateNicknameUseCase,
+) : ViewModel() {
+    private val _currentPage: MutableLiveData<OnBoardingPage> = MutableLiveData(OnBoardingPage.FIRST)
+    val currentPage: LiveData<OnBoardingPage> = _currentPage
+
+    private val _progressBarPercent: MutableLiveData<Int> = MutableLiveData(OnBoardingPage.FIRST.progressPercent)
+    val progressBarPercent: LiveData<Int> = _progressBarPercent
+
+    private val _isBackButtonVisible: MutableLiveData<Boolean> = MutableLiveData(OnBoardingPage.FIRST.isBackButtonVisible)
+    val isBackButtonVisible: LiveData<Boolean> = _isBackButtonVisible
+
+    private val _isSkipTextVisible: MutableLiveData<Boolean> = MutableLiveData(OnBoardingPage.FIRST.isSkipTextVisible)
+    val isSkipTextVisible: LiveData<Boolean> = _isSkipTextVisible
+
+    private val _onBoardingFirstUiState: MutableLiveData<OnBoardingFirstUiState> = MutableLiveData(
+        OnBoardingFirstUiState(
+            nicknameInputType = NicknameInputType.INITIAL,
+            nicknameValidationMessage = "",
+            isDuplicationCheckButtonEnable = false,
+            isNextButtonEnable = false
+        )
+    )
+    val onBoardingFirstUiState: LiveData<OnBoardingFirstUiState> = _onBoardingFirstUiState
+
+    val currentNicknameInput: MutableLiveData<String> = MutableLiveData("")
+
+    private fun updateUIBasedOnPage(page: OnBoardingPage) {
+        _progressBarPercent.value = page.progressPercent
+        _isBackButtonVisible.value = page.isBackButtonVisible
+        _isSkipTextVisible.value = page.isSkipTextVisible
+    }
+
+    fun validateNickname() {
+        val currentInput: String = currentNicknameInput.value.orEmpty()
+        if (currentInput.isEmpty()) {
+            updateOnBoardingFirstUiState(NicknameInputType.INITIAL, "")
+        } else {
+            val (isSuccess, message) = validateNicknameUseCase(currentInput)
+            if (isSuccess) {
+                updateOnBoardingFirstUiState(NicknameInputType.TYPING, message)
+            } else {
+                updateOnBoardingFirstUiState(NicknameInputType.ERROR, message)
+            }
+        }
+    }
```

Comment:
```suggestion
    fun validateNickname() {
        val currentInput: String = currentNicknameInput.value.orEmpty()
        if (currentInput.isEmpty()) {
            updateOnBoardingFirstUiState(NicknameInputType.INITIAL, "")
            return
        }
        val (isSuccess, message) = validateNicknameUseCase(currentInput)
        when (isSuccess) {
            true -> updateOnBoardingFirstUiState(NicknameInputType.TYPING, message)
            false -> updateOnBoardingFirstUiState(NicknameInputType.ERROR, message)
        }
    }
```
이친구 너무 장풍 쏘는거같아서...! 이런 느낌은 어떨까요?